daffodil-runtime1-unparser/src/main/scala/edu/illinois/ncsa/daffodil/processors/unparsers/BCDUnparsers.scala

```diff
@@ -54,7 +54,14 @@ final class BCDIntegerMinLengthInBytesUnparser(
   extends BCDIntegerBaseUnparser(e) {
 
   override def getBitLength(state: ParseOrUnparseState): Int = {
-    val len = state.currentNode.get.asSimple.dataValue.asInstanceOf[Array[Byte]].length * 8
+    val bitLen = state.currentNode.get.asSimple.dataValue.asInstanceOf[JBigInteger].bitLength
```

Comment:
BigInteger.bitLength returns the length in bits in two's complement. But don't we want the length in bits in BCD? Do we need to unparse the BigInteger to a byte array first, and if that isn't at least minLength then add some padding bits? Or perhaps we pass the minLength into the bcdFromBigInteger() to accept a minLength, and the padding logic could be done there if it's easier?

Reply:
It might be worth adding some tdml:unparserTestCase cases to ensure that minLength is behaving correctly. This feels to me like maybe we're adding more pad characters than is necessary and things just work because extra pad chars is okay?